lib/transformers/DockerCompose.js

```diff
@@ -239,16 +245,31 @@ DockerComposeTransformer.prototype.transformHandlers = function() {
     return Q.all(promises);
 };
 
-DockerComposeTransformer.prototype.injectCompositionVars = function(content) {
-    if (!this.opts.compositionVars || this.opts.compositionVars.length === 0) {
-        return content;
+DockerComposeTransformer.prototype.injectCompositionVars = function() {
+    return injectVarsToObject(this.data, this.opts.compositionVars);
+};
+
+var injectVarsToObject = function(object, vars) {
+
+    function iterate(object) {
+        for (var property in object) {
+            if (object.hasOwnProperty(property)) {
+                if (typeof object[property] == "object")
+                    iterate(object[property]);
+                else {
+                    var field = object[property];
+                    var replaceWith = _.find(vars, {key: field});
```

Comment:
I think this needs a reverse look up.
At the moment you're testing to see if the value is one of the vars, but the can contain the var